### PR TITLE
Add ListClusterInfo interface

### PR DESCRIPTION
**Pull request by monadbobo**
to return the ip address and data directory of the whole cluster (including meta and storage).

### CODE REVIEW

File: src/common/interface/common.thrift [hunk 1]
```diff
@@ -197,6 +197,16 @@ struct LogInfo {
     2: TermID term_id;
 }
 
+struct DirInfo {
+    1: binary                   root,
```
Comment (laura-ding) [inline src/common/interface/common.thrift:201]: add comment for the root

File: src/common/interface/common.thrift [hunk 2]
```diff
@@ -308,6 +326,11 @@ enum ErrorCode {
     E_RESTORE_FAILURE                 = -2068,
     E_SESSION_NOT_FOUND               = -2069,
 
+    // ListClusterInfo Failure
+    E_LIST_CLUSTER_FAILURE              = -2070,
+    E_LIST_CLUSTER_GET_ABS_PATH_FAILURE = -2071,
```
Comment (panda-sheep) [inline src/common/interface/common.thrift:331]: Maybe `E_FAILED_GET_ABS_PATH ` and `E_LIST_CLUSTER_GET_ABS_PATH_FAILURE   `can be combined into one?

Comment (monadbobo) [reply src/common/interface/common.thrift:331]: Here I wanted to make it more intuitive for BR tool to see the error message.

Comment (panda-sheep) [reply src/common/interface/common.thrift:331]: ok

**Review by panda-sheep — APPROVED** [review]
Good job!

**Review by critical27 — APPROVED** [review]

**Pull request merged**